go/vt/vtgate/evalengine/eval_numeric.go

```diff
@@ -111,9 +111,9 @@ func evalToNumeric(e eval) evalNumeric {
 	case *evalJSON:
 		switch e.Type() {
 		case json.TypeTrue:
-			return newEvalBool(true)
+			return &evalFloat{f: 1.0}
```

Comment:
Turns out that converting a JSON boolean turns it into a 1.0 or 0.0 `float` value and not an `int64` 1 or 0. This was not covered by tests before.